store/cachekv/store.go

```diff
@@ -93,6 +93,32 @@ func (store *Store) Delete(key []byte) {
 	store.setCacheValue(key, nil, true)
 }
 
+type cEntry struct {
+	key string
+	val *cValue
+}
+
+func (store *Store) deleteCaches() {
+	if len(store.cache) > 100_000 {
```

Comment:
Any access, read or write, to any field of the Store type, including store.cache, requires the caller to take the store mutex.